build.gradle

```diff
@@ -2638,6 +2638,7 @@ project(':streams') {
     testImplementation libs.mockitoCore
     testImplementation libs.mockitoJunitJupiter // supports MockitoExtension
     testImplementation libs.junitPlatformSuiteEngine // supports suite test
+    testImplementation libs.opentelemetryProto
```

Comment:
Needed to add this test dependency explicitly.  The clients module has this dependency shaded, but it's the package is renamed from `io.opentelemetry.proto` to  `org.apache.kafka.shaded.io.opentelemetry.proto`. IntelliJ finds this fine, but my local gradle build can't so I need to add it.

Reply:
@bbejeck This doesn't seems right.

We should use `org.apache.kafka.shaded.io.opentelemetry.proto.metrics.v1.MetricsData` as dependency in `KafkaStreamsTelemetryIntegrationTest`.  And change `streams` project dependency in build.gradle from `api project(':clients')` to `api project(path: ':clients', configuration: 'shadow')`.

I have verified the change by taking your PR and applying the fixes mentioned above. You need to amend import-control to allow dependency.

Also prease verify the contents of the streams jar as well, regarding what all is included. You can do so by issuing `jar -tf kafka-streams.jar  > jar_tf_output` (pre and post). Also check the manifest file.

Reply:
Thanks, @apoorvmittal10 this definitely seems better I've updated it.